src/BlockTypes/OrderConfirmation/AbstractOrderConfirmationBlock.php

```diff
@@ -103,13 +103,8 @@ protected function get_order() {
 	/**
 	 * View mode for order details based on the order, current user, and settings.
 	 *
-	 * Possible values are:
-	 * - "full" user can view all order details.
-	 * - "limited" user can view some order details, but no PII. This may happen for example, if the user checked out as a guest.
-	 * - false user cannot view order details.
-	 *
 	 * @param \WC_Order|null $order Order object.
-	 * @return "full"|"limited"|false
+	 * @return string|false Returns "full" if the user can view all order details. False if they can view no details.
```

Comment:
I noticed we're returning a mix of string and false ("full" or false). Is there a particular reason why we cannot use a boolean return type here? Looking at the changes below, I only see that we're calling `( ! $permission )`. I cannot see a part where we're checking against `full`.

If we'd use a boolean, we could simplify the following two functions.

```php
// Before
if ( $this->is_customer_order( $order ) ) {
  return $this->is_current_customer_order( $order ) ? 'full' : false;
}

// After
if ( $this->is_customer_order( $order ) ) {
  return ! $this->is_current_customer_order( $order );
}
```

```php
// Before
return $this->email_verification_required( $order ) ? false : 'full';

// After
return $this->email_verification_required( $order );
```

Reply:
I didn't want to introduce a breaking change. We allowed full, limited, or false perviously. I was not certain this class was being extended by 3rd parties or not.

Reply:
Thanks for explaining this, Mike. I haven't thought about folks potentially extending this class.